hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -127,13 +111,15 @@ private void createSymbolicLink() {
           log.info("Unable to unlink old sym link. Link may not exist. Exit code: " + exitCode);
         }
       } catch (IOException e) {
-        log.fatal("Could not unlink " + hdfsBinaryPath + ": " + e);
-        System.exit(1);
+        log.fatal("Could not unlink " + hdfsBinaryPath, e);
+        throw e;
```

Comment:
Ah, I see...it is surrounded by a second try catch, which then exits upon exception. This does in fact make sense. I think this block should be moved into a separate method to avoid complexity. Future PR :)